python/nano/src/bigdl/nano/pytorch/lightning.py

```diff
@@ -56,30 +64,31 @@ def training_step(self, batch, batch_idx):
 
     def validation_step(self, batch, batch_idx):
         y_hat = self._forward(batch)
-        loss = self.loss(y_hat, batch[-1])  # use last output as target
-        self.log("val/loss", loss, on_epoch=True, prog_bar=True, logger=True)
+        if self.loss:
+            loss = self.loss(y_hat, batch[-1])  # use last output as target
+            self.log("val/loss", loss, on_epoch=True,
+                     prog_bar=True, logger=True)
         if self.metrics:
-            acc = {"val/" + type(metric).__name__: metric(y_hat, batch[-1])
+            acc = {"val/{}_{}".format(type(metric).__name__, i): metric(y_hat, batch[-1])
                    for i, metric in enumerate(self.metrics)}
             self.log_dict(acc, on_epoch=True, prog_bar=True, logger=True)
-        else:
-            acc = None
-        return loss, acc
 
     def test_step(self, batch, batch_idx):
         y_hat = self._forward(batch)
-        loss = self.loss(y_hat, batch[-1])  # use last output as target
-        self.log("test/loss", loss, on_epoch=True, prog_bar=True, logger=True)
         if self.metrics:
-            acc = {"test/" + type(metric).__name__: metric(y_hat, batch[-1])
+            acc = {"test/{}_{}".format(type(metric).__name__, i): metric(y_hat, batch[-1])
                    for i, metric in enumerate(self.metrics)}
             self.log_dict(acc, on_epoch=True, prog_bar=True, logger=True)
-        else:
-            acc = None
-        return loss, acc
+
+    def predict_step(self, batch, batch_idx):
+        return self(*batch)
 
     def configure_optimizers(self):
-        return self.optimizer
+        optimizers = [self.optimizer]
```

Comment:
Why this change? This PR should focus on OpenCV support; you may submit a separate PR for this change if needed.